mux.go

```diff
@@ -312,6 +312,9 @@ func Vars(r *http.Request) map[string]string {
 }
 
 // CurrentRoute returns the matched route for the current request, if any.
+// Note: this only works when called inside the handler of the matched route
```

Comment:
Change the wording a bit. Remove "Note:" and instead of "it uses context.Get()" say:

"the matched route is stored in the request context which is cleared after the handler returns, unless the `KeepContext` option is set on the Router.`